sympy/core/tests/test_relational.py

```diff
@@ -661,3 +661,12 @@ def test_issue_10304():
     assert d.is_comparable is False  # if this fails, find a new d
     e = 1 + d*I
     assert simplify(Eq(e, 0)) is S.false
+
+
+def test_issue_8571():
+    raises(TypeError, lambda: x*S.true)
+    raises(TypeError, lambda: x + S.true)
+    raises(TypeError, lambda: S.true + x)
+    raises(TypeError, lambda: S(2)*S.false)
+    raises(TypeError, lambda: S(3)*S(True))
+    raises(TypeError, lambda: S(True)*S(3.0))
```

Comment:
1. Does this deal with `S(1)*S.false` (which behaves differently currently)?  If so, add test for that too.
 
2. Use with `S.true` or `S(True)` but don't mix here.